shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.osv import expression
+
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+from odoo.addons.shopfloor.utils import to_float
+
+
+class Reception(Component):
+    _inherit = "shopfloor.reception"
+
+    @property
+    def dimension_to_update(self):
+        if not hasattr(self.work, "dimension_to_update"):
+            self.work.dimension_to_update = {}
+        return self.work.dimension_to_update
+
+    def _before_response_for_set_quantity(self, picking, line):
+        """Show the packaging dimension screen before the set quantity screen."""
+        if not self.work.menu.set_packaging_dimension:
+            return self._response_for_set_quantity(picking, line)
+        packaging = self._get_next_packaging_to_set_dimension(line.product_id)
+        if packaging:
+            return self._response_for_set_packaging_dimension(picking, line, packaging)
+        return self._response_for_set_quantity(picking, line)
+
+    def _get_domain_packaging_needs_dimension(self):
+        return expression.OR(
+            [
+                [("packaging_length", "=", 0)],
+                [("packaging_length", "=", False)],
+                [("width", "=", 0)],
+                [("width", "=", False)],
+                [("height", "=", 0)],
+                [("height", "=", False)],
+                [("max_weight", "=", 0)],
+                [("max_weight", "=", False)],
+                [("qty", "=", 0)],
+                [("qty", "=", False)],
+                [("barcode", "=", False)],
+            ]
+        )
+
+    def _get_next_packaging_to_set_dimension(self, product, previous_packaging=None):
+        """Return for a product the next packaging needing dimension to be set."""
+        next_packaging_id = previous_packaging.id + 1 if previous_packaging else 0
+        domain_dimension = self._get_domain_packaging_needs_dimension()
+        domain_packaging_id = [
+            ("product_id", "=", product.id),
+            ("id", ">=", next_packaging_id),
+        ]
+        domain = expression.AND([domain_packaging_id, domain_dimension])
+        return self.env["product.packaging"].search(domain, order="id", limit=1)
+
+    def _response_for_set_packaging_dimension(
+        self, picking, line, packaging=None, message=None
+    ):
+        if not packaging:
+            packaging = self._get_next_packaging_to_set_dimension(line.product_id)
+        if not packaging:
```

Comment:
do we really need this check and call for something that should happen before?

Reply:
Indeed the packaging parameter should not be optional, and the ifs can be removed.